### PR TITLE
fix(streaming): clear cache if necessary when scale

### PR DESCRIPTION
**Pull request by BugenZhao**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

The problem is quite subtle and explained in #5567. For now, we implement a temporal fix mentioned [here](https://github.com/risingwavelabs/risingwave/issues/5567#issuecomment-1274610690).

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
- #5567

### CODE REVIEW

**Comment by BugenZhao** [comment]
I'm trying to add a scale test that can reproduce the issue before this PR. 🤔

File: src/stream/src/executor/top_n/utils.rs [hunk 1]
```diff
@@ -106,6 +109,12 @@ where
                 Message::Chunk(chunk) => yield Message::Chunk(self.inner.apply_chunk(chunk).await?),
                 Message::Barrier(barrier) => {
                     self.inner.flush_data(barrier.epoch).await?;
+
+                    // Update the vnode bitmap, only used by Group Top-N.
```
Comment (BugenZhao) [inline src/stream/src/executor/top_n/utils.rs:113]: This is forgotten in #4621.

**Review by fuyufjh — APPROVED** [review]
LGTM

**Comment by yuhao-su** [comment]
LGTM

**Pull request merged**